src/components/service/serviceUtils.ts

```diff
@@ -17,13 +17,21 @@ export const parseStorageKey = (input: string): StorageKey => {
     if (input.length === 64) {
       return bytes.Bytes.parseBytes(input, 31);
     }
+    if (input.length === 50) {
+      const paddedInput = input + '0'.repeat(16);
+      return bytes.Bytes.parseBytes(paddedInput, 32);
+    }
   }
   const hasher = blake2b(32);
   hasher.update(bytes.BytesBlob.blobFromString(input).raw);
   return bytes.Bytes.fromBlob(hasher.digest(), 32);
 };
 
 export const parsePreimageHash = (input: string): PreimageHash => {
+  if (input.startsWith('0x') && input.length === 50) {
+    const paddedInput = input + '0'.repeat(16);
+    return bytes.Bytes.parseBytes(paddedInput, 32);
+  }
```

Comment:
Here the logic needs to be different. We support either full 66-characters (`0x` + 64-characters) and turn that into `Bytes<32>` or specify 64 characters (`0x` + 62 characters) and turn it into `Bytes<31>`. In the second case we assume this is the storage key already, so it doesn't have to go through state access (i.e. `service.getPreimage`) but rather should query the raw state directly to fetch the value.